src/SLCore.UnitTests/Common/Models/FileUriTests.cs

```diff
@@ -114,6 +114,19 @@ public void ToString_PercentEncodesBackticks()
         new FileUri(@"C:\filewithbacktick`1").ToString().Should().Be("file:///C:/filewithbacktick%601");
     }
 
+    [TestMethod]
+    [DataRow("[", "%5B")]
+    [DataRow("]", "%5D")]
+    [DataRow("#", "%2523")]
+    [DataRow("@", "%40")]
+    public void ToString_PercentEncodesReservedRfc3986Characters(string reservedChar, string expectedEncoding)
```

Comment:
Maybe add similar tests, but in reverse: from `file://...` to FileUri.LocalPath, rather than from filePath to uri string